src/main/java/atdd/station/controller/LineController.java

```diff
@@ -0,0 +1,93 @@
+package atdd.station.controller;
+
+import atdd.station.exception.ErrorType;
+import atdd.station.exception.SubwayException;
+import atdd.station.model.CreateEdgeRequestView;
+import atdd.station.model.CreateLineRequestView;
+import atdd.station.model.dto.LineDto;
+import atdd.station.model.entity.Edge;
+import atdd.station.model.entity.Line;
+import atdd.station.service.LineService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/lines")
+public class LineController {
+    @Autowired
+    private LineService lineService;
+
+    @PostMapping
+    public ResponseEntity<LineDto> createLine(@RequestBody CreateLineRequestView view) {
+        final Line line = lineService.create(view.toLine());
+
+        LineDto lineDto = lineService.lineToLineDto(line);
+
+        return ResponseEntity.created(URI.create("/lines/" + line.getId()))
+                .body(lineDto);
+    }
+
+    @GetMapping
+    public ResponseEntity<List<LineDto>> findAllLines() {
+        final List<Line> lines = lineService.findAll();
+
+        List<LineDto> lineDtos = new ArrayList<>();
+
+        for (Line line : lines)
+            lineDtos.add(lineService.lineToLineDto(line));
+
+        return ResponseEntity.ok(lineDtos);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<LineDto> findLine(@PathVariable long id) throws SubwayException {
+        final Optional<Line> optionalLine = lineService.findById(id);
+
+        if (optionalLine.isPresent()) {
+            LineDto lineDto = lineService.lineToLineDto(optionalLine.get());
+
+            return ResponseEntity.ok(lineDto);
+        }
+
+        throw new SubwayException(ErrorType.NOT_FOUND_LINE);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity deleteLine(@PathVariable long id) {
+        lineService.deleteById(id);
+
+        return ResponseEntity.noContent().build();
+    }
+
+    @PostMapping("/{id}/edge")
+    public ResponseEntity<LineDto> addEdge(@PathVariable long id,
+                                           @RequestBody CreateEdgeRequestView view) throws SubwayException {
+        final Edge.EdgeBuilder edgeBuilder = Edge.builder()
+                .sourceStationId(view.getSourceStationId())
+                .targetStationId(view.getTargetStationId());
+
+        Optional<Line> lineOptional = lineService.addEdge(id, edgeBuilder.build());
```

Comment:
```suggestion
        Line line = lineService.addEdge(id, view.toEntity());
```

1. Builder 자료형을 별도로 노출할 필요가 있을까요? 
2. getter를 사용할 경우, 객체지향적인지 한번 생각해볼 필요가 있을거 같아요.
    getter를 사용한다는 것은, 그 외부에서 해당 객체 내부의 속성을 알아야 한다는 의미인데요.
    이렇게 내부 데이터에 초점을 두고 프로그래밍 할 경우에 내부의 변화가 외부로 전파될 경우가 발생하게 됩니다.
    위의 로직을 View 내부로 이동하는 것은 어떨까요
3. Optional을 비용이 비싼 작업입니다. Optional을 사용하지 않는다면, 위 메서드의 내부 구현은 어떻게 작성하는 것이 좋을까요
>http://homoefficio.github.io/2019/10/03/Java-Optional-%EB%B0%94%EB%A5%B4%EA%B2%8C-%EC%93%B0%EA%B8%B0/